katcp/resource_client.py

```diff
@@ -1111,45 +1170,77 @@ def list_sensors(self, filter="", strategy=False, status="",
             dict.items(self.sensor), filter, strategy, status, use_python_identifiers, tuple, refresh)
 
     @tornado.gen.coroutine
-    def set_sampling_strategy(self, resource_name, sensor_name, strategy_and_parms):
-        sensor_name_in = sensor_name
-        sensor_name = resource.escape_name(sensor_name)
-        if not sensor_name.startswith("agg_"):
-            # Set strategy on resource client - which will cache it if necessary
-            resource_obj = self.children[resource_name]
-            yield resource_obj.set_sampling_strategy(sensor_name, strategy_and_parms)
-        else:
-            # Handle aggregate sensors that are not alwasy pre-allocated to the same mon_ component
-            # TODO: Handle aggregates better
-            # (for now the aggregate sensor_obj must exist as you don't know on which resource to cache it)
-            sensor_obj = getattr(self.sensor, sensor_name, None)
-            if sensor_obj:
-                resource_obj = self.children[sensor_obj.parent_name]
+    def _resource_set_sampling_strategies(self, resource_name, sensor_name, strategy_and_parms):
```

Comment:
Ok - always forget that one as it is superficial to me. But then, rules are rules.